### PR TITLE
[codemod] Add missing `preset-safe` for the data-grid

### PR DESCRIPTION
**Pull request by arminmeh**
Adds missing `preset-safe` for the data grid
https://github.com/mui/mui-x/pull/15659 helped detect that

I have added pickers section to the readme

Also, I have aligned formatting for the diffs in the readme (spacing for changed and unchanged lines)

### CODE REVIEW

File: packages/x-codemod/README.md [hunk 1]
```diff
@@ -95,31 +97,31 @@ The list includes these transformers
 Renames the Tree View component to Simple Tree View
 
 ```diff
--import { TreeView } from '@mui/x-tree-view';
-+import { SimpleTreeView } from '@mui/x-tree-view';
+- import { TreeView } from '@mui/x-tree-view';
```
Comment (LukasTy) [inline packages/x-codemod/README.md:100]: Olivier has recently fixed such formatting issues. 🙈 
https://github.com/mui/mui-x/pull/15575#discussion_r1855318679

File: packages/x-codemod/README.md [hunk 2]
```diff
@@ -212,17 +214,51 @@ The list includes these transformers
...
+  />
 ```
 
 ```bash
 npx @mui/x-codemod@latest v8.0.0/data-grid/remove-stabilized-experimentalFeatures <path>
 ```
 
+### Pickers codemods
+
+#### `preset-safe` for pickers v8.0.0
```
Comment (LukasTy) [inline packages/x-codemod/README.md:230]: Nitpick: I think that the products should start with upper case. 🤔 
```suggestion
#### `preset-safe` for Pickers v8.0.0
```

File: packages/x-codemod/README.md [hunk 3]
```diff
@@ -212,17 +214,51 @@ The list includes these transformers
...
+
+The `preset-safe` codemods for pickers.
```
Comment (LukasTy) [inline packages/x-codemod/README.md:232]:
```suggestion
The `preset-safe` codemods for Pickers.
```

File: packages/x-codemod/README.md [hunk 4]
```diff
@@ -212,17 +214,51 @@ The list includes these transformers
...
+
+```bash
+npx @mui/x-codemod@latest v8.0.0/pickers/preset-safe <path|folder>
```
Comment (LukasTy) [inline packages/x-codemod/README.md:235]: Technically this should be `next` for now on the `master` branch. 🤔 
But in that case, we need to add the comment to not forget to change it. 🙈 

Screenshot 2024-12-03 at 11 12 15


```suggestion
<!-- #default-branch-switch -->
npx @mui/x-codemod@next v8.0.0/pickers/preset-safe <path|folder>
```

File: packages/x-codemod/README.md [hunk 5]
```diff
@@ -212,17 +214,51 @@ The list includes these transformers
...
+  interface MyComponentProps {
+-   valueType: FieldValueType;
++   valueType: PickerValueType;
+    foo: string;
+    bar: number;
+  }
+```
+
+```bash
+npx @mui/x-codemod@latest v8.0.0/pickers/rename-and-move-field-value-type <path>
```
Comment (LukasTy) [inline packages/x-codemod/README.md:259]:
```suggestion
<!-- #default-branch-switch -->
npx @mui/x-codemod@next v8.0.0/pickers/rename-and-move-field-value-type <path>
```

**Review by LukasTy — COMMENTED** [review]
Nice refinement. 👍 

P.S. WDYT about using the `codemod` prefix for the PR?
It directly impacts the package. 🤔

File: packages/x-codemod/README.md (hunk 1 above)
Comment (arminmeh) [reply packages/x-codemod/README.md:100]: Ah ok, so it is the other way around :)
I'll update this

**Comment by arminmeh** [comment]
> P.S. WDYT about using the codemod prefix for the PR?
It directly impacts the package. 🤔

I have used the linked PR as a reference (same situation), I don't mind having it either way

File: packages/x-codemod/README.md (hunk 4 above)
Comment (arminmeh) [reply packages/x-codemod/README.md:235]: Didn't know about the comment. Thanks

**Review by LukasTy — APPROVED** [review]
LGTM. 👌 
Thanks for the adjustments. 💯 🙏

**Pull request merged**